packages/api/src/controllers/animalGroupController.js

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import baseController from './baseController.js';
+import AnimalGroupModel from '../models/animalGroupModel.js';
+import AnimalGroupRelationshipModel from '../models/animalGroupRelationshipModel.js';
+import AnimalBatchGroupRelationshipModel from '../models/animalBatchGroupRelationshipModel.js';
+import AnimalModel from '../models/animalModel.js';
+
+const animalGroupController = {
+  addAnimalGroup() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      try {
+        const { farm_id } = req.headers;
+        const { related_animal_ids, related_batch_ids } = req.body;
+        let { name } = req.body;
+        name = baseController.checkAndTrimString(name);
+
+        if (!name) {
+          await trx.rollback();
+          return res.status(400).send('Group name is required');
+        }
+
+        if (!Array.isArray(related_animal_ids) || !Array.isArray(related_batch_ids)) {
+          await trx.rollback();
+          return res.status(400).send('Animal ids and batch ids must be arrays');
+        }
+
+        if (!related_animal_ids.length && !related_batch_ids.length) {
+          await trx.rollback();
+          return res.status(400).send('Either animal ids or batch ids are required');
+        }
```

Comment:
We might want to allow creation of an empty group depending on how the UI works -- I could imagine a scenario where you can create the group with its name and leave it empty in a sort of "draft" state to add animals and batches later

Reply:
Done, it will accept both arrays as empty now.